packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentStackTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, stackName: string, stackSdk: ISDK) : Promise<Template> {
```

Comment:
I feel like this method should switch places in the file with `prepareSdkForLookup()`, as `prepareSdkForLookup()` is called first in all paths that I can see that lead to `readCurrentStackTemplate()`.